cmd/clusterctl/client/cluster/client.go

```diff
@@ -38,14 +38,27 @@ var (
 	ctx = context.TODO()
 )
 
+// Kubeconfig is a type that specifies inputs related to the actual
+// kubeconfig.
+type Kubeconfig struct {
+	// Path to the kubeconfig file
+	Path string
+	// Specify context within the kubeconfig file. If empty, cluster client
+	// will use the current context.
+	Context string
+}
+
 // Client is used to interact with a management cluster.
 // A management cluster contains following categories of objects:
 // - provider components (e.g. the CRDs, controllers, RBAC)
 // - provider inventory items (e.g. the list of installed providers/versions)
 // - provider objects (e.g. clusters, AWS clusters, machines etc.)
 type Client interface {
-	// Kubeconfig return the path to kubeconfig used to access to a management cluster.
-	Kubeconfig() string
+	// KubeconfigPath return the path to kubeconfig used to access to a management cluster.
+	KubeconfigPath() string
```

Comment:
```suggestion
	Kubeconfig() Kubeconfig
```
So we avoid adding a getter whenever the struct changes?

Reply:
Ahh yes. I wanted to do this. Thanks for bringing this up.